cmd/rpcdaemon/commands/eth_api.go

```diff
@@ -329,50 +331,53 @@ func newRPCTransaction(tx types.Transaction, blockHash libcommon.Hash, blockNumb
 	// because the return value of ChainId is zero for those transactions.
 	chainId := uint256.NewInt(0)
 	result := &RPCTransaction{
-		Type:  hexutil.Uint64(tx.Type()),
-		Gas:   hexutil.Uint64(tx.GetGas()),
-		Hash:  tx.Hash(),
-		Input: hexutil.Bytes(tx.GetData()),
-		Nonce: hexutil.Uint64(tx.GetNonce()),
-		To:    tx.GetTo(),
-		Value: (*hexutil.Big)(tx.GetValue().ToBig()),
+		Type:                hexutil.Uint64(tx.Type()),
+		Gas:                 hexutil.Uint64(tx.GetGas()),
+		Hash:                tx.Hash(),
+		Input:               hexutil.Bytes(tx.GetData()),
+		Nonce:               hexutil.Uint64(tx.GetNonce()),
+		To:                  tx.GetTo(),
+		BlobVersionedHashes: tx.GetDataHashes(),
+		Value:               (*hexutil.Big)(tx.GetValue().ToBig()),
 	}
+
+	chainId.Set(tx.GetChainID())
+	result.ChainID = (*hexutil.Big)(chainId.ToBig())
```

Comment:
nit: This could be merged into `result :=` block

Reply:
ah yes that's better, done.